### PR TITLE
@wdio/allure-reporter: Skip reporting of passing hooks with option useCucumberStepReporter

### PR DESCRIPTION
**Pull request by Kignuf**
## Proposed changes
When using the allure reporter option "useCucumberstepReporter", this change will stop reporting passing hooks, and reattach any attachment from those hooks to the test-case.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments
I have checked "Breaking change" because this does change the way allure reporter behaves when using "useCucumberReporter" option, though I believe it's for the better ;)

### CODE REVIEW

File: packages/wdio-allure-reporter/src/index.js [hunk 1]
```diff
@@ -228,6 +228,16 @@ class AllureReporter extends WDIOReporter {
             // remove hook from suite if it has no steps
             if (this.allure.getCurrentTest().steps.length === 0 && !this.options.useCucumberStepReporter) {
                 this.allure.getCurrentSuite().testcases.pop()
+            } else if (this.options.useCucumberStepReporter) {
+                // remove hook when it's registered as a step and if it's passed
+                const step = this.allure.getCurrentTest().steps.pop()
+
+                // if it had any attachments, reattach them to current test
+                if (step.attachments.length >= 1) {
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/src/index.js]:
```suggestion
                if (step && step.attachments.length >= 1) {
```

Comment (Kignuf) [reply packages/wdio-allure-reporter/src/index.js]: Fixed :)

**Comment by mgrybyk** [comment]
Hey guys, I'm working on some changes in hooks, may you please hold on on merging this one, I want to verify it in my branch, will try to do ASAP.

https://github.com/webdriverio/webdriverio/pull/4288

**Comment by mgrybyk** [comment]
@Kignuf what should I do in order to get into `} else if (this.options.useCucumberStepReporter) {` ? 
I have defined all the hooks and they are not appearing in the allure report

**Comment by mgrybyk** [comment]
ok, works fine with my branch!

**Comment by Kignuf** [comment]
Passing hooks are not shown on the report anymore, that's the expected result.
If you want to see hooks on the report, make one of them fail.

**Pull request merged**